dash/development/base_component.py

```diff
@@ -392,6 +402,14 @@ def __repr__(self):
             props_string = repr(getattr(self, "children", None))
         return f"{self._type}({props_string})"
 
+    @classmethod
+    def _get_base_nodes(cls):
+        if not cls._base_nodes:
+            cls._base_nodes = [
+                n.replace("[]", "").split(".")[0] for n in cls._children_props
```

Comment:
Could this result in duplicate entries in this list, like if `a.b` and `a.c` both support components? Does that matter?

Reply:
It's only used for a check, better change it to a set.

Reply:
actually this is used just in the [type check](https://github.com/plotly/dash/pull/1965/files#diff-ac4d356fec919ee468c3b73bbd242cddf1deb95b985a6b860e44a79176117211R143), right? That isn't digging into nested prop vals, so I'd think we only want the props that support components as their top level - ie children props that don't have `[]` or `.` in them - otherwise you could provide a component at the top level for that prop, bypass this error and only get the error on the front end.

(and a small optimization: might as well add `'children'` to the list here rather than in the type check, and in fact only include it if the component has `'children'` in its `_prop_names` - or `available_properties`, IDK why we have both of these, maybe we should drop `_prop_names` and just use `available_properties`, with a fallback in the constructor to move `_prop_names` over to `available_properties` if some old component only has `_prop_names`?)

Reply:
Good point, should only check the flat props on the backend.

Reply:
> `available_properties` is generated in the init, I think they should be class members like the children props as it's static data.

Good point. But if we change that we'll need a test to ensure old generated components still work as expected. I don't imagine there would be a problem but the old components will be in use for a long time.